app/services/provider_authorisation.rb

```diff
@@ -129,23 +133,34 @@ def full_authorisation?(permission:, course:)
     # enforce user-level permissions
     errors << :requires_provider_user_permission unless
       @actor.is_a?(VendorApiUser) ||
-        @actor.provider_permissions.send(permission)
-          .exists?(provider: [training_provider, ratifying_provider].compact)
+        user_level_can?(permission: permission, provider: training_provider) ||
+        user_level_can?(permission: permission, provider: ratifying_provider)
 
     # enforce org-level permissions
     if ratifying_provider.present?
```

Comment:
What's the case where there is no ratifying provider - does that mean the course is ratified by the training provider? Why don't we need to enforce org-level permissions in this case?

Reply:
In that case there is no relationship to consider